app/src/main/java/com/android/unio/ui/event/EventCreation.kt

```diff
@@ -1,20 +1,379 @@
+        readOnly = true,
+        onValueChange = {},
+        trailingIcon = { Icon(Icons.Default.DateRange, contentDescription = "Select date") },
+        placeholder = { Text(context.getString(R.string.event_creation_placeholder_date_input)) },
+        label = { Text(dateString) })
+    Spacer(modifier = Modifier.weight(0.05f))
+    OutlinedTextField(
+        modifier =
+            Modifier.weight(1f).pointerInput(Unit) {
+              awaitEachGesture {
+                // Modifier.clickable doesn't work for text fields, so we use Modifier.pointerInput
+                // in the Initial pass to observe events before the text field consumes them
+                // in the Main pass.
+                awaitFirstDown(pass = PointerEventPass.Initial)
+                val upEvent = waitForUpOrCancellation(pass = PointerEventPass.Initial)
+                if (upEvent != null) {
+                  isTimePickerVisible = true
+                }
+              }
+            },
+        value = selectedTime?.let { convertMillisToTime(it) } ?: "",
+        readOnly = true,
+        onValueChange = {},
+        trailingIcon = { Icon(Icons.Default.AccessTime, contentDescription = "Select date") },
+        placeholder = { Text(context.getString(R.string.event_creation_placeholder_time_input)) },
+        label = { Text(timeString) })
+  }
+
+  if (isDatePickerVisible) {
+    DatePickerModal(
+        onDateSelected = {
+          selectedDate = it
+          isDatePickerVisible = false
+        },
+        onDismiss = { isDatePickerVisible = false })
   }
+
+  if (isTimePickerVisible) {
+    TimePickerModal(
+        onTimeSelected = {
+          selectedTime = it
+          isTimePickerVisible = false
+        },
+        onDismiss = { isTimePickerVisible = false })
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DatePickerModal(onDateSelected: (Long?) -> Unit, onDismiss: () -> Unit) {
+  val datePickerState = rememberDatePickerState()
+  val context = LocalContext.current
+
+  DatePickerDialog(
+      onDismissRequest = onDismiss,
+      confirmButton = {
+        TextButton(
+            onClick = {
+              onDateSelected(datePickerState.selectedDateMillis)
+              onDismiss()
+            }) {
+              Text(context.getString(R.string.event_creation_dialog_ok))
+            }
+      },
+      dismissButton = {
+        TextButton(onClick = onDismiss) {
+          Text(context.getString(R.string.event_creation_dialog_cancel))
+        }
+      }) {
+        DatePicker(state = datePickerState)
+      }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TimePickerModal(onTimeSelected: (Long?) -> Unit, onDismiss: () -> Unit) {
+  val timePickerState = rememberTimePickerState(is24Hour = true)
+
+  TimePickerDialog(
+      onDismiss = onDismiss,
+      onConfirm = {
+        onTimeSelected(
+            timePickerState.hour * 60 * 60 * 1000L - 3600000 + timePickerState.minute * 60 * 1000L)
```

Comment:
Consider using `TimeUnit` to make these conversions more readable, e.g. to go from an hour value to a ms value, use `TimeUnit.HOURS.toMillis()`